crates/bevy_math/src/curve/mod.rs

```diff
@@ -968,19 +971,135 @@ where
     }
 }
 
+/// Convert a `start` and `end` point of some [`VectorSpace`] type into a [`Curve`] over the
+/// [unit interval], sampled by linearly interpolating between the two boundary values.
+///
+/// [unit interval]: `Interval::UNIT`
+pub fn line_curve<T: VectorSpace>(start: T, end: T) -> impl Curve<T> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| start * (1.0 - t) + end * t,
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = x²`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn quadratic_ease_in() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| t.squared(),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = 1 - (1 - x)²`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn quadratic_ease_out() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| 1.0 - (1.0 - t).squared(),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(x) = x² * (3 - 2x)`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn cubic_curve() -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: |t: f32| t.squared() * (3.0 - 2.0 * t),
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(n,x) = round(x * n) / n`
+///
+/// parametrized by `n`, the number of steps
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn step_curve(num_steps: usize) -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| (t * num_steps as f32).round() / num_steps as f32,
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function `f(omega,x) = (1 - (1 - x)²)(2sin(omega * x) / omega + cos(omega * x))`
+///
+/// parametrized by `omega`
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn elastic_curve(omega: f32) -> impl Curve<f32> {
+    FunctionCurve {
+        domain: Interval::UNIT,
+        f: move |t: f32| {
+            (1.0 - (1.0 - t).squared()) * (2.0 * ops::sin(omega * t) / omega + ops::cos(omega * t))
+        },
+        _phantom: PhantomData,
+    }
+}
+
+/// A [`Curve`] mapping the [unit interval] to itself.
+///
+/// It uses the function
+/// `f(p0,p1,p2,p3,x) = (1 - x)³ * p0 + 3(1 - x)² * x * p1 + 3(1 - x) * x² * p2 + t³ * p3`
+///
+/// parametrized by `p0`, `p1`, `p2`, `p3`, called "control points"
+///
+/// [unit domain]: `Interval::UNIT`
+pub fn cubic_bezier_curve(control_points: [f32; 4]) -> impl Curve<f32> {
```

Comment:
Not a fan of effectively duplicating `CubicSegment<f32>` here, although I'm not against this helper method returning a `CubicSegment<f32>`.

One reason is that CubicSegment has a (imho) better implementation for the cubic polynomial, by rearranging the coefficients during construction (based on expanding out all the parentheses in your given expression) and then simply evaluating `A + tB + t^2 C + t^3 D`.

Also I'd again note, that unless the user chooses p3 = 1.0 and p0 = 0.0 and (if I'm correct) also 0 <= p1, p2 <= 1, then the curve does not map the unit interval to itself. (If we demand a one-to-one bijective mapping, then further constraints may apply on p1 and p2, though I'm not sure there.)

Reply:
Totally reasonable. I kind of didn't think about it. I'll just create a Curve struct as a wrapper around the `CubicSegement<T:VectorSpace>` :) Thanks for catching that. It feels like a better API which is already documented for free which is really nice!